### ISSUE TITLE
Too many SOQL queries: 101 

### ISSUE THREAD

**Issue by tamilust** [afawcett/declarative-lookup-rollup-summaries#276]
I am getting a SOQL error when trying to deploy a new trigger. Do you know why I would be getting this error and point me in the direction to fixing it.

System.LimitException: dlrs:Too many SOQL queries: 101 
Stack Trace: (dlrs) Trigger.dlrs_ContactTrigger: line 7, column 1

Thank you,

Tami

**Comment by afawcett** [comment]
What other triggers do you have on the Contact?
Do you have any other rollups on Contact?

**Comment by tamilust** [comment]
I have a few other rollups on the Contact object. I checked them all to make sure the correct names are in the rollups. Should I be checking for something else? There are other triggers on Contact but nothing new.

**Comment by afawcett** [comment]
This maybe a case of trigger recursion, what does your rollup definition look like? Is Contact the parent and child? Would any of the triggers attempt to update other Contact records?

**Comment by tamilust** [comment]
This rollup is deploying a new trigger. When trying to deploy the trigger I get an error on the Contact class. When you ask about the contact information do you mean on all the rollups I have for the contact record? This trigger that I am trying to deploy is not on the contact record. I do not have any rollups that the contact is the parent and child.

**Comment by afawcett** [comment]
Is it possible for you to list the steps to reproduce and then grant subscriber support access for me to see this first hand? What i ment earlier was to see your rollup definitions setup in the tool.

**Comment by tamilust** [comment]
I have granted login access to my org for 30 days. 

Steps to reproduce:
1.Navigate to this rollup: https://na9.salesforce.com/a1JE0000006mVG8
2. Try and deploy the trigger
3. It will fail and you will see the error: Test_ContactTrigger.testContactTrigger System.LimitException: dlrs:Too many SOQL queries: 101 (dlrs) Trigger.dlrs_ContactTrigger: line 7, column 1

Thank you.

**Comment by tamilust** [comment]
Have you had a chance to log into the org and investigate the issue?

Thank you

Tami

**Comment by afawcett** [comment]
No, i thought i had the Org ID, but i don't, can you provide please.

**Comment by tamilust** [comment]
No problem. Org ID: 00DE0000000YOIC


Tami Lust
Abel Cine NY
609 Greenwich Street, 4th - 5th Floors New York, NY 10014
888.223.1599 | 212.462.0175 |
tlust@abelcine.com<mailto:tlust@abelcine.com> | www.abelcine.com<http://www.abelcine.com/>
New York • Chicago • Los Angeles

On Jan 19, 2016, at 2:58 PM, Andrew Fawcett <notifications@github.com<mailto:notifications@github.com>> wrote:


No, i thought i had the Org ID, but i don't, can you provide please.

**Comment by tamilust** [comment]
Any update on this?

**Comment by afawcett** [comment]
I get the following errors....

MyProfilePageController.testSave System.DmlException: Insert failed. First exception on row 0; first error: CANNOT_EXECUTE_FLOW_TRIGGER, The record couldn’t be saved because it failed to trigger a flow. A flow trigger failed to execute the flow with version ID 301E0000000bs9v. Contact your administrator for help.: [] Class.MyProfilePageController.testSave: line 93, column 1

Test_ContactTrigger.testContactTrigger System.LimitException: dlrs:Too many SOQL queries: 101 (dlrs) Trigger.dlrs_ContactTrigger: line 7, column 1

Having looked at your huge list of rollups, i'm afraid to say i think your really pushing the realtime limits of the rollups. Salesforce imposes a limit of 100 queries per package (in this case dlrs) per invocation (e.g. a set of records being updated on Contact). 

If you have say circa 32 +/-  of these, all with their own criteria, the tool cannot help put run 32 separate queries for this (it tries its best to combine rollups into a single query but not if the criteria is different). Now 32 is of course less than 100, however... you do have over a dozen triggers on Contact and possibly some Workflows? I cannot tell, because most are from packages and the code is not visible, however if any one of those causes a recursive update (one Contact causes an update to another Contact), this 32 can easily get multiplied. Also if you happen to touch the Account records, which in turn appear to have many many rollups defined this will also burn queries. 

So in short i can only recommend you really try to rationalise your rollups on Account and Contact, and maybe consider some of them to be moved to Scheduled mode.

**Comment by tamilust** [comment]
Thank you for the clear explanation. I have moved a bunch of jobs on the contact object to scheduled mode and I was able to deploy the trigger. I have tried scheduled mode in the past and it didn't seem like it ever ran. When does scheduled mode run? Is there a way for me to set the schedule? Thanks again for shedding light on this issue.

**Comment by afawcett** [comment]
Take a look at this blog for details on how to set the time your scheduled rollups are processed. http://andyinthecloud.com/2014/02/09/new-release-spring14-declarative-rollup-summary-tool/ Thanks! :+1:
